aspnetcore/razor-pages/rp-mvc.md

```diff
@@ -0,0 +1,56 @@
+
+* Can continue app development featuring controllers and views.
+* Can be assured the ASP.NET Core framework will provide improvements to controller/view development.
+
+Razor Pages can be added to controller/view projects. The controller-view approach to MVC may be preferred by larger teams.
+
+If you haven't tried Razor Pages, consider using them. Many developers find Razor Pages development simpler and more productive than using controllers and views.
+
+Razor Pages advantages over controller/view development:
+
+* **Leads to more maintainable code and a folder structure by default that's easier to navigate.**  
+  * Controllers often end up with a large number of `Action` methods, many of which are not related to each other.
+  * Controllers contain all the shared code used by all the `Action` methods. Razor Pages include only the code they use. Shared code for Razor Pages is contained in a base class. For example, see [Create a base class to share common code](xref:data/ef-rp/update-related-data#create-a-base-class-to-share-common-code).
+
+   MVC uses separate folders for Views, Controllers, and ViewModels when all three are tightly coupled. You must bounce between all three folders when you need to add/change/debug a feature. With Razor Pages, the `PageModel` (Controller + ViewModel) is in the same folder as the View.  ~Hit F7 in Visual Studio to toggle between them.~
+
+   Razor Pages follow the [Single responsibility principle](https://wikipedia.org/wiki/Single_responsibility_principle), while MVC does not.
+* **Unit Testing is easier.**  With a Controller, you might have many Actions and some of the dependencies injected that are related to only one or two Actions.  When unit testing a single Action, the dependencies must be mocked or passed in as null. With Razor Pages, the dependencies you inject in are 100% related to GET, POST, PUT, etc actions in the `PageModel`.  Unnecessary dependency injection in MVC testing can be mitigated with the [Builder pattern](https://visualstudiomagazine.com/articles/2012/07/16/the-builder-pattern-in-net.aspx).
+* **Routing is easier.**  By default in Razor Pages, routing matches the folder structure.  This makes nesting folders straightforward.  For example, consider an app where:
+  * All of the HR administrator/privileged pages are under the */Administrator* folder.
+  * All the Employee pages are under the */Employee* folder.  
+  * The app can authorize an entire folder and require the user to be an Administrator to get to any subfolder of */Administrator*. The code to do this is more straightforward than using multiple Controllers for Administrator features.
+* **More secure by default.**  Razor Pages provides:
+  * [AntiForgeryToken validation](xref:razor-pages/index#xsrfcsrf-and-razor-pages) by default.
+  * You opt-in to the properties that need to be model bound via `[BindProperty]`.  `[BindProperty]` prevents over-posting attacks.
```

Comment:
Maybe provide a link to over posting attacks in case someone reading has no clue what this is.  [OWASP uses the term Mass Assignment](https://www.owasp.org/index.php/Mass_Assignment_Cheat_Sheet) although their example is in Spring + Java.

Not sure if you want to link to external blogs or not but Hanselman has a good example here that's MVC specific: https://www.hanselman.com/blog/ASPNETOverpostingMassAssignmentModelBindingSecurity.aspx and Scott Allen has a good one too: https://odetocode.com/Blogs/scott/archive/2012/03/11/complete-guide-to-mass-assignment-in-asp-net-mvc.aspx